jwst/emicorr/emicorr.py

```diff
@@ -0,0 +1,792 @@
+        Sigma for rejection
+
+    maxiter: int
+        Maximum number of sigma rejection iterations
+
+    Returns
+    -------
+    dmean : float
+        Computed mean
+
+    dmedian : float
+        Computed median
+
+    dsigma : float
+        Computed sigma
+
+    dmask : numpy array
+        Mask set to 1 for good points, and 0 for rejected points
+    """
+    # special cases of 0 or 1 data points
+    ngood = np.size(data)
+    dmean, dmedian, dsigma, dmask = 0.0, 0.0, 0.0, np.zeros(np.shape(data))
+    if ngood == 0:
+        log.debug('No data points for sigma clipping')
+        return dmean, dmedian, dsigma, dmask
+    elif ngood == 1:
+        log.debug('Only 1 data point for sigma clipping')
+        return dmean, dmedian, dsigma, dmask
+
+    # Compute the mean + standard deviation of the entire data array,
+    # these values will be returned if there are fewer than 2 good points.
+    dmask = np.ones(ngood, dtype='b') + 1
+    dmean = sum(data * dmask) / ngood
+    dsigma = np.sqrt(sum((data - dmean)**2) / (ngood - 1))
+    dsigma = dsigma
+    iiter = 1
+
+    # Iteratively compute the mean + stdev, updating the sigma-rejection thresholds
+    # each iteration
+    nlast = -1
+    while iiter < maxiter and nlast != ngood and ngood >= 2:
+        loval = dmean - sigrej * dsigma
+        hival = dmean + sigrej * dsigma
+        nlast = ngood
+
+        dmask[data < loval] = 0
+        dmask[data > hival] = 0
+        ngood = sum(dmask)
+
+        if ngood >= 2:
+            dmean = sum(data*dmask) / ngood
+            dsigma = np.sqrt( sum((data - dmean)**2 * dmask) / (ngood - 1) )
+            dsigma = dsigma
+
+        iiter += 1
+
+    return dmean, dmedian, dsigma, dmask
+
+
+def rebin(arr, newshape):
+    """Rebin an array to a new shape.
+
+    Parameters
+    ----------
+    arr : numpy array
+        array to rebin
+
+    newshape : list
+        New shape for the input array
+
+    Returns
+    -------
+    arr : numpy array
+        rebinned array
+    """
+    assert len(arr.shape) == len(newshape)
+
+    slices = [ slice(0,old, float(old)/new) for old,new in zip(arr.shape,newshape) ]
+    coordinates = np.mgrid[slices]
+    indices = coordinates.astype('i')   # choose the biggest smaller integer index
+
+    return arr[tuple(indices)]
+
+
+def mk_reffile(freq_pa_dict, emicorr_ref_filename):
+    """ Create the reference file hdulist object.
+
+    Parameters
+    ----------
+    freq_pa_dict : dictionary
+        Dictionary containing the phase amplitude (pa) array
+        corresponding to the appropriate frequency
+
+    emicorr_ref_filename : str
+        Full path and root name of on-the-fly reference file
+
+    Returns
+    -------
+    Nothing
+    """
+    # save the reference file if save_intermediate_results=True and no CRDS file exists
+    emicorr_model = datamodels.EmiModel(freq_pa_dict)
+    if 'fits' in emicorr_ref_filename:
+        emicorr_ref_filename = emicorr_ref_filename.replace('fits', 'asdf')
```

Comment:
Is this going to automatically change a user-specified file name that contains a ".fits" extension into ".asdf" instead, hence forcing the output to always be in ASDF format?

Reply:
nope, this was a work around because I used the Step class method _make_output_path to return the full path and name of an output file (it always returns with .fits)